### PR TITLE
C++ fixes for mac build

### PR DESCRIPTION
**Pull request by lunixbochs**
These are pulled from @realdoug's changes in https://github.com/facebookresearch/wav2letter/issues/222

After this, we should just need a fix or workaround for distributed mode to support wav2letter on mac upstream.

### CODE REVIEW

File: src/common/Transforms.h [hunk 1]
```diff
@@ -145,7 +145,7 @@ std::vector<T> localNormalize(
     int64_t curFrame = 0;
     // accumulate sum, sum^2 for computing mean, stddev
     for (auto i = b * perBatchSz; i < (b + 1) * perBatchSz; ++i) {
-      auto start = std::max(curFrame - rightCtxSize, 0L);
+      auto start = std::max(curFrame - rightCtxSize, INT64_C(0));
```
Comment (jacobkahn) [inline src/common/Transforms.h]: prefer `int64_t(0)`

Comment (jacobkahn) [inline src/common/Transforms.h]: prefer `int64_t(0)`

File: src/common/Transforms.h [hunk 2]
```diff
@@ -156,7 +156,7 @@ std::vector<T> localNormalize(
     // compute mean, stddev
     for (auto j = 0; j < frameSz; ++j) {
       int64_t N = (std::min(j + rightCtxSize, frameSz - 1) -
-                   std::max(j - leftCtxSize, 0L) + 1) *
+                   std::max(j - leftCtxSize, INT64_C(0)) + 1) *
```
Comment (jacobkahn) [inline src/common/Transforms.h]: ditto

**Comment by lunixbochs** [comment]
Ok, I made the int64_t changes (didn't know that was an available macro)

**Comment by lunixbochs** [comment]
What's the status on the OpenMPI backend? I'd like to stop rebasing my fork sometime :)

**Comment by lunixbochs** [comment]
Seems like this is unlikely to land, let us know if you ever add mac support upstream.

**Comment by jacobkahn** [comment]
Will do. There are some other unfortunate blockers to MPI stand-alone support that will probably require removing MPI altogether at some point. This probably won’t be revisited for a while. Thanks again for the work.

**Comment by lunixbochs** [comment]
Reminder that Mac doesn't really have nvidia support, so distributed training on Mac support is pretty pointless. Perhaps reconsider the original changes to make distributed mode optional in wav2letter.

**Pull request closed without merging**